frame/support/src/traits/misc.rs

```diff
@@ -950,6 +950,24 @@ impl<Hash> PreimageRecipient<Hash> for () {
 	fn unnote_preimage(_: &Hash) {}
 }
 
+/// Trait to handle asset locking mechanism to ensure interactions with the asset can be implemented
+/// downstream to extend logic of Uniques current functionality
```

Comment:
```suggestion
/// downstream to extend logic of Uniques current functionality.
```

Reply:
If this trait is only useful for the uniques pallet, it should be in `token` folder, not `misc.rs`.

Reply:
Would `token/misc.rs` be the best place?